src/hansken-js.js

```diff
@@ -1,47 +1,93 @@
-     * @param {String} keystoreUrl The url to the Hansken keystore
+     * @param {string} gatekeeperUrl The url to the Hansken gatekeeper
+     * @param {string} keystoreUrl The url to the Hansken keystore
      */
     constructor(gatekeeperUrl, keystoreUrl) {
         this.sessionManager = new SessionManager(gatekeeperUrl, keystoreUrl);
     }
 
     /**
-     * Get all projects.
+     * Create a new project.
      *
-     * @returns All projects the current user is authorized for
+     * @param {object} project The project as specified in the REST API docs.
+     * @returns A ProjectContext for the new project
      */
-    projects = () => this.sessionManager.gatekeeper('/projects').then((response) => response.json());
+    createProject = (project) => this.sessionManager.gatekeeper(`/projects`, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(project)
+    }).then(SessionManager.parseLocationId)
+    .then(this.project);
 
     /**
-     * Get all single files.
+     * Get a context for a single project, to do project specific REST calls.
      *
-     * @returns All single files the current user is authorized for
+     * @param {UUID} projectId The project id
+     * @returns A ProjectContext for a project
      */
-    singlefiles = () => this.sessionManager.gatekeeper('/singlefiles').then((response) => response.json());
+    project = (projectId) => new ProjectContext(this.sessionManager, projectId);
 
     /**
-     * Get a context for a single project, to do project specific REST calls.
+     * Get all projects.
      *
-     * @param {UUID} projectId The project id
-     * @returns A ProjectContext for a project
+     * @returns All projects the current user is authorized for
      */
-    project = (projectId) => new ProjectContext(this.sessionManager, 'projects', projectId);
+    projects = () => this.sessionManager.gatekeeper(`/projects`).then((response) => response.json());
+
+    /**
+     * Create a new singlefile.
+     *
+     * @param {string} name The name for the singlefile
+     * @param {Blob | BufferSource | ReadableStream} data The data of the singlefile
+     * @returns The singlefile id
+     */
+    createSinglefile = (name, data) => this.sessionManager.gatekeeper(`/singlefiles/upload/${name}`, {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/octet-stream'
+        },
+        body: data
+    }).then(SessionManager.parseLocationId);
 
     /**
      * Get a context for a single file, to do project specific REST calls.
      *
      * @param {UUID} singlefileId The single file id
      * @returns A ProjectContext for a single file
      */
-    singlefile = (singlefileId) => new ProjectContext(this.sessionManager, 'singlefiles', singlefileId);
+    singlefile = (singlefileId) => new SinglefileContext(this.sessionManager, singlefileId);
+
+    /**
+     * Get all single files.
```

Comment:
Do we have an opinion on whether it's a "singlefile" or a "single file" (and therefore a `singlefile` or a `singleFile`). If so, we could make it a bit more consistent.

Reply:
I don't like the name singlefile either, the inventors of it explained to me that singlefile is the name, not single fil

Reply:
Hansken code bases uses `Singlefile` everywhere as 1 word